addons/survey/tests/test_survey.py

```diff
@@ -296,3 +297,44 @@ def get_question_by_title(survey, title):
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_question_id.id, q_is_vegetarian.id)
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_answer_id.id,
                             q_is_vegetarian.suggested_answer_ids[1].id)
+
+    def test_get_questions_except_invalid_conditionals(self):
+        """ Tests the method `_get_questions_except_invalid_conditionals`. This method takes a
+            recordset of question.question from a survey.survey and returns a recordset without
+            conditional questions where the condition is not valid. """
+
+        survey_feedback = self.env.ref('survey.survey_feedback')
+        passed_questions = survey_feedback.question_and_page_ids.filtered(
+            lambda question: not question.is_page or not is_html_empty(question.description)).sorted()
+        returned_questions = survey_feedback._get_questions_except_invalid_conditionals(passed_questions)
+
+        # All questions are valid at the start of the survey
+        self.assertEqual(passed_questions, returned_questions)
+
+        # Valid conditional question
+        p2_q1 = self.env.ref('survey.survey_feedback_p2_q1')
+        p2_q1.write({
+            'is_conditional': True,
+            'triggering_question_id': self.env.ref('survey.survey_feedback_p1_q3'),
+            'triggering_answer_id': self.env.ref('survey.survey_feedback_p1_q3_sug1'),
+        })
+        returned_questions = survey_feedback._get_questions_except_invalid_conditionals(passed_questions)
+        self.assertEqual(passed_questions, returned_questions)
+
+        # Invalid conditional question
+        p2_q1.write({'sequence': 0})
+        passed_questions = passed_questions.sorted()
+        returned_questions = survey_feedback._get_questions_except_invalid_conditionals(passed_questions)
+        self.assertEqual(len(passed_questions) - 1, len(returned_questions))
+        self.assertTrue(p2_q1 not in returned_questions)
```

Comment:
You can also use this to be more specific and straightforward doing two in one.
```suggestion
        self.assertEqual(passed_questions - p2q1, returned_questions)
```